store/multiversion/mvkv.go

```diff
@@ -122,6 +123,14 @@ func (store *VersionIndexedStore) GetWriteset() map[string][]byte {
 	return store.writeset
 }
 
+// WriteAbort writes an abort to the store but only allows one abort to be written PER instance of mvkv. This is because we pair abort channel writes with panics, and if we hit this more than once, it means that the panic was swallowed, so we won't write any aborts after a first abort is written to prevent any potential for deadlocking due to full channels
+func (store *VersionIndexedStore) WriteAbort(abort scheduler.Abort) {
+	if !store.abortWritten {
```

Comment:
unless there's a lock situation, i see this racing

Reply:
```
select {
    case ch <- 2: // Put 2 in the channel unless it is full
    default:
        fmt.Println("Channel full. Discarding value")
    }
```
could we try this kind of shape?  This would prevent a 'full' situation at least

Reply:
updated to gracefully not write to channel